### PR TITLE
Fundamentals index update

### LINKED ISSUE #7000: Why the docs uses the namespace `aspnetcoreapp` in lowercase instead of `AspNetCoreApp` in PascalCase?
Doesn't the [names conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces) says to use PascalCase in namespaces names? The example for .NET Core 2.x uses this:

```c#
namespace aspnetcoreapp
```

Why is it different here?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a7e622e9-c567-00b9-3788-d6a01934b314
* Version Independent ID: 8e663029-63b9-2622-f852-9f8a2ec70db3
* Content: [ASP.NET Core fundamentals](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/index?view=aspnetcore-2.1&tabs=aspnetcore2x)
* Content Source: [aspnetcore/fundamentals/index.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/index.md)
* Product: **asp.net-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5495 
Fixes #7000

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/index?view=aspnetcore-1.0&branch=pr-en-us-8187)

* Let's avoid ranges and a sample app that isn't linked for download by using snapshots of the two files needed for this topic.
* Little text enhancements throughout.
* Convert to xref API links everywhere.
* Swap moniker ranges for tabs.
* Include tomorrow's winning lottery numbers.
* Update topic links to xref format (pull the titles).

### CODE REVIEW

File: aspnetcore/fundamentals/index.md [hunk 1]
```diff
@@ -1,216 +1,206 @@
...
 
 ## Startup
 
 The `UseStartup` method on `WebHostBuilder` specifies the `Startup` class for your app:
 
-# [ASP.NET Core 2.x](#tab/aspnetcore2x/)
+::: moniker range=">= aspnetcore-2.0"
 
-[!code-csharp[](../getting-started/sample/aspnetcoreapp/Program2x.cs?highlight=10&range=6-17)]
+[!code-csharp[](index/snapshots/2.x/Program.cs?highlight=10)]
```
Comment (scottaddie) [inline aspnetcore/fundamentals/index.md]: A *samples* folder should be used to store the sample code.

File: aspnetcore/fundamentals/index.md [hunk 2]
```diff
@@ -1,216 +1,206 @@
...
+<xref:Microsoft.AspNetCore.Hosting.IStartup.ConfigureServices*> defines the [Services](#dependency-injection-services) used by your app (for example, ASP.NET Core MVC, Entity Framework Core, Identity). <xref:Microsoft.AspNetCore.Hosting.IStartup.Configure*> defines the [middleware](xref:fundamentals/middleware/index) called in the request pipeline.
 
-For more information, see [Application startup](xref:fundamentals/startup).
+For more information, see <xref:fundamentals/startup>.
 
 ## Content root
 
-The content root is the base path to any content used by the app, such as views, [Razor Pages](xref:razor-pages/index), and static assets. By default, the content root is the same as application base path for the executable hosting the app.
+The content root is the base path to any content used by the app, such as [Razor Pages](xref:razor-pages/index), MVC views, and static assets. By default, the content root is the same location as app base path for the executable hosting the app.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/index.md]: ~~same location as app~~
same location as the app

File: aspnetcore/fundamentals/index.md [hunk 3]
```diff
@@ -1,216 +1,206 @@
...
+
+## Hosting
+
+ASP.NET Core apps configure and launch a *host*, which is responsible for app startup and lifetime management.
+
+For more information, see <xref:fundamentals/host/index>.
+
+## Servers
+
+The ASP.NET Core hosting model doesn't directly listen for requests. The hosting model relies on an HTTP server implementation to forward the request to the app. The forwarded request is wrapped as a set of feature objects that can be accessed through interfaces. ASP.NET Core includes a managed, cross-platform web server, called [Kestrel](xref:fundamentals/servers/kestrel). Kestrel is commonly run behind a production web server, such as [IIS](https://www.iis.net/) or [Nginx](http://nginx.org) in a reverse proxy configuration. Kestrel can also be run as an edge server exposed directly to the Internet.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/index.md]: I believe this last sentence is false for 2.0 and below. Or, at least, it wasn't recommended until the 2.1 release came along.

File: aspnetcore/fundamentals/index.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/index.md]: There's no runnable, downloadable sample for this topic. It's purely example code; therefore, a snapshot of the two files is all that's required.

File: aspnetcore/fundamentals/index.md (hunk 3 above)
Comment (guardrex) [reply aspnetcore/fundamentals/index.md]: Yes! I'll fix it.

**Pull request merged**